src/ray/gcs/gcs_client/accessor.cc

```diff
@@ -459,9 +459,9 @@ void ActorInfoAccessor::AsyncResubscribe() {
   // server first, then fetch data from the GCS server.
   absl::MutexLock lock(&mutex_);
   for (auto &[actor_id, resubscribe_op] : resubscribe_operations_) {
-    RAY_CHECK_OK(resubscribe_op([this, actor_id = actor_id](const Status &status) {
+    RAY_CHECK_OK(resubscribe_op([this, id = actor_id](const Status &status) {
```

Comment:
```suggestion
    RAY_CHECK_OK(resubscribe_op([this, actor_id](const Status &status) {
```
can you just try this instead, it'll do the same copy into the capture and should avoid the shadow warning

Reply:
Thanks @dayshah I'll fix it.

The microcheck fails because many other shadowing fixes are not pushed yet (over 1600 warnings). I plan to split to 2~3 PRs for easier review.

Reply:
Is ray using c++20? captured structure binding is a c++ 20 feature
![image](https://github.com/user-attachments/assets/911dbdfb-e90c-4f21-a605-bce04a34a468)

Reply:
oh it's actually not using 20, still on 17. My bad didn't realize it was a 20 only thing, can go back to what you originally had